src/app/core/data/dspace-rest-v2-response-parsing.service.ts

```diff
@@ -0,0 +1,265 @@
+import { hasNoValue, hasValue, isNotEmpty } from '../../shared/empty.util';
+import { DSpaceSerializer } from '../dspace-rest-v2/dspace.serializer';
+import { CacheableObject } from '../cache/object-cache.reducer';
+import { Serializer } from '../serializer';
+import { PageInfo } from '../shared/page-info.model';
+import { ObjectCacheService } from '../cache/object-cache.service';
+import { GenericConstructor } from '../shared/generic-constructor';
+import { PaginatedList, buildPaginatedList } from './paginated-list.model';
+import { getClassForType } from '../cache/builders/build-decorators';
+import { RestRequest } from './request.models';
+import { environment } from '../../../environments/environment';
+import { DSpaceRESTV2Response } from '../dspace-rest-v2/dspace-rest-v2-response.model';
+import { DSpaceObject } from '../shared/dspace-object.model';
+import { Injectable } from '@angular/core';
+import { ResponseParsingService } from './parsing.service';
+import { ParsedResponse } from '../cache/response.models';
+import { RestRequestMethod } from './rest-request-method';
+import { getUrlWithoutEmbedParams } from '../index/index.selectors';
+
+/* tslint:disable:max-classes-per-file */
+
+/**
+ * Return true if obj has a value for `_links.self`
+ *
+ * @param {any} obj The object to test
+ */
+export function isCacheableObject(obj: any): boolean {
+  return hasValue(obj) && hasValue(obj._links) && hasValue(obj._links.self) && hasValue(obj._links.self.href);
+}
+
+/**
+ * Return true if halObj has a value for `page` with properties
+ * `size`, `totalElements`, `totalPages`, `number`
+ *
+ * @param {any} halObj The object to test
+ */
+export function isRestPaginatedList(halObj: any): boolean {
+  return hasValue(halObj.page) &&
+    hasValue(halObj.page.size) &&
+    hasValue(halObj.page.totalElements) &&
+    hasValue(halObj.page.totalPages) &&
+    hasValue(halObj.page.number);
+}
+
+/**
+ * Split a url into parts
+ *
+ * @param url the url to split
+ */
+const splitUrlInParts = (url: string): string[] => {
+  return url.split('?')
+    .map((part) => part.split('&'))
+    .reduce((combined, current) => [...combined, ...current])
+}
+
+@Injectable({ providedIn: 'root' })
+export class DSpaceRESTV2ResponseParsingService implements ResponseParsingService {
```

Comment:
Since this is a huge PR that touches all the same files that a separate rename PR would touch, we might as well rename them now. What name would you prefer? Simply `DSpaceRest…` ?

Reply:
@artlowel : Yes, simply `DSpaceRest` is fine by me.  I just don't like the `V2` as it doesn't make sense for a version 7 of DSpace :)

Reply:
Done